pkg/pxc/app/proxysqlcnf/cluster.go

```diff
@@ -0,0 +1,123 @@
+package proxysqlcnf
+
+import (
+	"context"
+	"fmt"
+	api "github.com/Percona-Lab/percona-xtradb-cluster-operator/pkg/apis/pxc/v1alpha1"
+	"github.com/Percona-Lab/percona-xtradb-cluster-operator/pkg/pxc/app/statefulset"
+	"github.com/pkg/errors"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"time"
+)
+
+type ClusterManager struct {
+	cli client.Client
+}
+
+func NewClusterManager(client client.Client) *ClusterManager {
+	return &ClusterManager{
+		cli: client,
+	}
+}
+
+func (c *ClusterManager) InitiateCluster(cr *api.PerconaXtraDBCluster) error {
+	proxyMembers, err := c.fetchProxyMembers(cr)
+	if err != nil {
+		return errors.Wrap(err, "failed to initiate the cluster")
+	}
+
+	if len(proxyMembers.Items) < 2 {
+		return errors.New("can't initialize the cluster. Not enough proxysql nodes")
+	}
+
+	hostnameList, err := c.podsHostnameList(proxyMembers.Items)
+	if err != nil {
+		return errors.Wrap(err, "failed to initiate the cluster")
+	}
+
+	for _, proxy := range proxyMembers.Items {
+
+		// get pod hostname
+		hostname, err := c.podHostname(proxy)
+		if err != nil {
+			return errors.Wrap(err, "failed to initiate the cluster")
+		}
+
+		// connect to proxysql node
+		db, err := NewDB(hostname)
```

Comment:
Is it enough to establish connection?

Reply:
It uses the connect method which uses Open and then Ping. So yeah that's pretty enough